sbi/inference/base.py

```diff
@@ -196,3 +193,73 @@ def _assert_all_finite(quantity: Tensor, description: str = "tensor") -> None:
 
         msg = f"NaN/Inf present in {description}."
         assert torch.isfinite(quantity).all(), msg
+
+    def _summarize(
+        self,
+        round_,
+        x_o,
+        theta_bank,
+        x_bank,
+        simulator,
+        posterior_samples_acceptance_rate=None,
+    ):
+        """Update the summary_writer with statistics for a given round.
+
+        Statistics are extracted from the arguments and from entries in self._summary
+        created during training.
+
+        XXX: This is a subset of the logging from the original Conor repository. A big
```

Comment:
This is a comment for developers so not for the docstring (=how to use) but as a code comment (=workflow state, implementation caveats, etc. Also, the code will be public, I feel more comfortable with just 'the original repository' or if you want, conormdurkan/lfi).